modules/core/src/main/java/org/apache/fluo/core/observer/v2/ObserverStoreV2.java

```diff
@@ -102,16 +104,16 @@ public RegisteredObservers load(CuratorFramework curator) throws Exception {
     String json = new String(data, UTF_8);
     JsonObservers jco = new Gson().fromJson(json, JsonObservers.class);
 
-    Set<Column> weakColumns = new HashSet<>();
-    Set<Column> strongColumns = new HashSet<>();
+    ImmutableSet.Builder<Column> weakColumnsBuilder = new ImmutableSet.Builder<Column>();
+    ImmutableSet.Builder<Column> strongColumnsBuilder = new ImmutableSet.Builder<Column>();
 
     for (Entry<Column, NotificationType> entry : jco.getObservedColumns().entrySet()) {
       switch (entry.getValue()) {
         case STRONG:
-          strongColumns.add(entry.getKey());
+          strongColumns = strongColumnsBuilder.add(entry.getKey()).build();
```

Comment:
Building the map for each iteration of the loop may be expensive.  I would suggest only adding in the loop and then building the map after the loop.